Add SRV records support to rdata

ClearlyClaire opened the pull request:
This adds support for SRV RR data. I am planning to implement the usage rules in domain::utils, but as I am new to Rust and tokio, this might take a while.

ClearlyClaire committed:
Add SRV records support to rdata

partim commented on src/rdata/mod.rs:
```diff
@@ -69,9 +73,11 @@ pub fn fmt_rdata(rtype: ::iana::Rtype, parser: &mut ::bits::Parser,
 pub mod parsed {
     pub use super::rfc1035::parsed::*;
     pub use super::rfc3596::Aaaa;
+    pub use super::rfc2782::Srv;
```
Since `Srv` is generic over the domain name type, this needs to be a type alias for `Srv<ParsedDName<'a>>`. We could either define that here or have a sub-module `rfc2782::parsed` just like `rfc1035` has. Since there’s only one record type defined in RFC 2782, I suppose defining a `pub type Srv<'a>` right here is fine.

partim commented on src/rdata/mod.rs:
```diff
@@ -69,9 +73,11 @@ pub fn fmt_rdata(rtype: ::iana::Rtype, parser: &mut ::bits::Parser,
 pub mod owned {
     pub use super::rfc1035::owned::*;
     pub use super::rfc3596::Aaaa;
+    pub use super::rfc2782::Srv;
```
As above; this one needs to be a type alias for `Srv<DNameBuf>`.

partim reviewed (commented):
Thank you very much for your PR. I added two inline comments—I will also add some explanatory comments to those two modules in a separate commit to make their purpose more clear.

As for implementing the usage rules, these should go into `domain::resolv::lookup`. Would be great if you started implementing it. Since SRV processing is relatively complex, it is a good test whether the underlying APIs are any good or need improvement. Feedback regarding the APIs themselves and documentation is very much appreciated!

ClearlyClaire commented:
I think I have addressed your remarks regarding the type aliases.
Thanks for the pointer, I'll look into implementing SRV usage rules in domain::resolv::lookup later this week!

partim commented:
Looks very good now, thank you!

If you have any issues with the resolver part, let me know!
